drivers/android/thread.rs

```diff
@@ -120,11 +124,13 @@ impl InnerThread {
     fn push_existing_work(&mut self, owork: Option<Arc<ThreadError>>, code: u32) {
         // TODO: Write some warning when the following fails. It should not happen, and
         // if it does, there is likely something wrong.
-        if let Some(mut work) = owork {
-            if let Some(work_mut) = Arc::get_mut(&mut work) {
-                work_mut.error_code = code;
-                self.push_work(work);
-            }
+        if let Some(work) = owork {
+            // `error_code` is written to with relaxed semantics because the queue onto which it is
+            // being inserted is protected by a lock. The release barrier when the lock is released
+            // by the caller matches with the acquire barrier of the future reader to guarantee
+            // that `error_code` is visible.
+            work.error_code.store(code, Ordering::Relaxed);
```

Comment:
Nit: does the `Relaxed` ordering really make a practical difference here? If not, is this perhaps a "micro-optimization", which serves to make the code more fragile - because `Relaxed` introduces a dependency on the queue lock ?

Reply:
I don't think it makes any difference as this code path is only taken on failure, which is very rare. I will update it to remove the relaxed store.

Reply:
As I was making the change I realised that this is also used when completing transactions with errors, for example, all pending transactions that a process has while exiting. We don't really want to unnecessarily slow things down in such cases.

So I'll leave this relaxed.